Core/src/org/sleuthkit/autopsy/centralrepository/eventlisteners/CaseEventListener.java

```diff
@@ -154,82 +206,90 @@ private ContentTagTask(CentralRepository db, PropertyChangeEvent evt) {
 
         @Override
         public void run() {
-            if (!CentralRepository.isEnabled()) {
+            if (!CentralRepository.isEnabled())
                 return;
-            }
 
-            AbstractFile af;
-            TskData.FileKnown knownStatus;
-            String comment;
             if (Case.Events.valueOf(event.getPropertyName()) == Case.Events.CONTENT_TAG_ADDED) {
-                // For added tags, we want to change the known status to BAD if the 
-                // tag that was just added is in the list of central repo tags.
-                final ContentTagAddedEvent tagAddedEvent = (ContentTagAddedEvent) event;
-                final ContentTag tagAdded = tagAddedEvent.getAddedTag();
-
-                if (TagsManager.getNotableTagDisplayNames().contains(tagAdded.getName().getDisplayName())) {
-                    if (tagAdded.getContent() instanceof AbstractFile) {
-                        af = (AbstractFile) tagAdded.getContent();
-                        knownStatus = TskData.FileKnown.BAD;
-                        comment = tagAdded.getComment();
-                    } else {
-                        LOGGER.log(Level.WARNING, "Error updating non-file object");
-                        return;
-                    }
-                } else {
-                    // The added tag isn't flagged as bad in central repo, so do nothing
-                    return;
-                }
-            } else { // CONTENT_TAG_DELETED
-                // For deleted tags, we want to set the file status to UNKNOWN if:
-                //   - The tag that was just removed is notable in central repo
-                //   - There are no remaining tags that are notable 
-                final ContentTagDeletedEvent tagDeletedEvent = (ContentTagDeletedEvent) event;
+                handleTagAdded((ContentTagAddedEvent) event);
+            } 
+            else { 
+                handleTagDeleted((ContentTagDeletedEvent) event);
+            }
+        }
+        
+        private void handleTagDeleted(ContentTagDeletedEvent tagDeletedEvent) {
+            // CONTENT_TAG_DELETED
+            // For deleted tags, we want to set the file status to UNKNOWN if:
+            //   - The tag that was just removed is notable in central repo
+            //   - There are no remaining tags that are notable 
+            //   - If there are other notable tags, the comment should be updated to most recent tag
+
+            if (!isNotableTagName(tagDeletedEvent.getDeletedTagInfo().getName())) {
+                // If the tag that got removed isn't on the list of central repo tags, do nothing
```

Comment:
Again, let's be careful with out comments. There is no such thing as a "central repo tag."